### PR TITLE
Feature restart

### PR DESCRIPTION
**Pull request by tnadrai**
restart simulation

### CODE REVIEW

**Comment by szerintedmi** [comment]
Deploy preview for *augmintsim* ready!

Built with commit 08f29cdc1b5303d94f9c90269d38a23a50933d6a

https://deploy-preview-76--augmintsim.netlify.com

File: src/augmint/augmint.js [hunk 1]
```diff
@@ -58,6 +58,82 @@ module.exports = {
...
+            ethToUsdTrend: 0
+        };
+
+        for (let i = this.orderBook.buy.length; i > 0; i--) {
+            self.orderBook.buy.pop();
+        }
+
+        for (let i = this.orderBook.sell.length; i > 0; i--) {
+            self.orderBook.sell.pop();
+        }
```
Comment (szerintedmi) [inline src/augmint/augmint.js]: could we just do `self.orderBook.buy.length = 0` (same for all arrays)? See [here](https://stackoverflow.com/questions/1232040/how-do-i-empty-an-array-in-javascript)

File: src/augmint/freezer.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const augmint = require("./augmint.js");
 const clock = require("../lib/clock.js");
 
 const ONE_DAY_IN_SECS = 24 * 60 * 60;
-const locks = augmint.locks;
+//const locks = augmint.locks;
```
Comment (szerintedmi) [inline src/augmint/freezer.js]: remove comment line?

File: src/lib/simulation.js [hunk 3]
```diff
@@ -33,6 +33,24 @@ function patchAugmintBalances(balances) {
...
+    // console.log("before clear");
+    // augmint.debugState();
+    // showLog();
+
+    iteration = 0;
+    clock.setTime(0);
+
+    for (var it = actors.values(), val = null; (val = it.next().value); ) {
+        if (val.locks.length > 0) {
+            console.log(val.id + ":" + JSON.stringify(val.locks[0].id));
```
Comment (szerintedmi) [inline src/lib/simulation.js]: remove console.log

File: src/index.html [hunk 4]
```diff
@@ -208,6 +209,7 @@
             <button class="btn clearLog-btn" onclick="javascript:document.getElementById('log-textarea').value = ''">clear log</button>
             <span>Dump to log:</span>
             <button class="btn dumpState-btn">state</button>
+            <button class="btn dumpIterationLog-btn">iteration log</button>
```
Comment (szerintedmi) [inline src/index.html]: I think I've removed iteration log dump button. maybe it sneaked back in a merge you did?

File: src/lib/simulation.js [hunk 5]
```diff
@@ -33,6 +33,24 @@ function patchAugmintBalances(balances) {
...
+            val.releaseACD(val.locks[0].id);
```
Comment (szerintedmi) [inline src/lib/simulation.js]: what are you trying to achieve here? why not just remove all actors?

File: src/ui.js [hunk 6]
```diff
@@ -318,14 +351,18 @@ function init() {
...
+
     pauseBtn.addEventListener("click", togglePause);
     ratesDropDown.addEventListener("change", () => ratesDropDownOnChange(ratesDropDown.value));
-
     dumpStateBtn.addEventListener("click", () => {
         simulation.patchAugmintParams(getParamsFromUI());
         logger.print(simulation.getState());
     });
-
+    dumpIterationLogBtn.addEventListener("click", () => {
```
Comment (szerintedmi) [inline src/ui.js]: same here: I think I've removed iteration log dump. maybe it sneaked back in a merge you did?

File: src/ui.js [hunk 7]
```diff
@@ -15,7 +14,9 @@ const ActorDirectory = require("./actors/actor.directory.js");
 // DOM elements
 const clockElem = document.querySelector(".clock");
 const pauseBtn = document.querySelector(".pause-btn");
+const restartBtn = document.querySelector(".restart-btn");
 const dumpStateBtn = document.querySelector(".dumpState-btn");
+const dumpIterationLogBtn = document.querySelector(".dumpIterationLog-btn");
```
Comment (szerintedmi) [inline src/ui.js]: same here: I think I've removed iteration log dump. maybe it sneaked back in a merge you did?

**Review by szerintedmi — CHANGES_REQUESTED** [review]

**Comment by szerintedmi** [comment]
@CryppTom : not all the change request has been addressed

File: src/lib/simulation.js (hunk 3 above)
Comment (tnadrai) [reply src/lib/simulation.js]: already removed in the last version: 
https://github.com/Augmint/augmint-sim/blob/feature_restart/src/lib/simulation.js

File: src/lib/simulation.js (hunk 5 above)
Comment (tnadrai) [reply src/lib/simulation.js]: already removed in the last version: 
https://github.com/Augmint/augmint-sim/blob/feature_restart/src/lib/simulation.js

File: src/augmint/augmint.js (hunk 1 above)
Comment (tnadrai) [reply src/augmint/augmint.js]: true, it's simpler, changed

File: src/augmint/freezer.js (hunk 2 above)
Comment (tnadrai) [reply src/augmint/freezer.js]: removed

File: src/ui.js (hunk 6 above)
Comment (tnadrai) [reply src/ui.js]: removed

File: src/ui.js (hunk 7 above)
Comment (tnadrai) [reply src/ui.js]: removed

**Review by szerintedmi — APPROVED** [review]

**Pull request merged**